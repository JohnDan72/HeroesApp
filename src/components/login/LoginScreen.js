import React from 'react';
import { Carousel, Col, CustomProvider, FlexboxGrid } from 'rsuite';
// import PropTypes from 'prop-types';

import FormLogin from './FormLogin';

// styles
import styles from './LoginScreen.module.css';



const LoginScreen = () => {

    return (
        <CustomProvider theme="dark">
            <FlexboxGrid className={styles.allHeighWidth} justify="center">
                <FlexboxGrid.Item className="h-100" as={Col} xsHidden smHidden md={12} lg={14} >
                    <Carousel className="h-100 custom-slider" autoplay >
                        <div className={styles.carousel1} >
                            <div className="v-a-middle allHeightWidth" justify="center" >
                                <h4> Heroes App </h4>
                            </div>
                        </div>
                        <div className={styles.carousel5} >
                            <div className="v-a-middle allHeightWidth" justify="center" >
                                <h4> Los héroes que todo el mundo conoce al alcance de un click </h4>
                            </div>
                        </div>
                        <div className={styles.carousel6} >
                            <div className="v-a-middle allHeightWidth" justify="center" >
                                <h4> Lo mejor de Marvel y DC </h4>
                            </div>
                        </div>
                        <div className={styles.carousel2} >
                            <div className="v-a-middle allHeightWidth"
                                justify="center" >
                                <h4> Busca y encuentra las características de tu héroe favorito </h4>
                            </div>
                        </div>
                    </Carousel>
                </FlexboxGrid.Item>
                <FlexboxGrid.Item as={Col} xs={22} md={12} lg={10} >
                    <FormLogin />
                </FlexboxGrid.Item>
            </FlexboxGrid>
        </CustomProvider>


    );
}


export default LoginScreen;