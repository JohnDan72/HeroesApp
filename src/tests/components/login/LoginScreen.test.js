import { mount } from "enzyme"
import { MemoryRouter } from "react-router";
import { GeneralContext } from "../../../GeneralContext";
import LoginScreen from "../../../components/login/LoginScreen";

describe('Pruebas LoginScreen', () => {
    global.fetch = jest.fn(() =>
        Promise.resolve({
            id: 1,
            name: 'JuanDan',
            email: 'juan100@gmail.com'
        })
    );

    const contextValue = {
        dispatch: jest.fn()
    }
    const wrapper = mount(
        <GeneralContext.Provider value={contextValue}>
            <MemoryRouter initialEntries={['/login']}>
                <LoginScreen />
            </MemoryRouter>
        </GeneralContext.Provider>
    );

    test('debe mostrar component correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    // TEST PENDIENTE
    // test('debe realizar el dispatch y la navegaciÃ³n ', () => {
    //     const emailEntry = 'otro@hotmail.com';
    //     console.log(wrapper.find(`input[id='id_email_input']`).prop('value'));
    //     wrapper.find(`input[id='id_email_input']`).simulate('change', { target: { value: emailEntry } });
    //     console.log(wrapper.find(`input[id='id_email_input']`).prop('value'));

    //     wrapper.find('Form').simulate('submit');

    // });


})
