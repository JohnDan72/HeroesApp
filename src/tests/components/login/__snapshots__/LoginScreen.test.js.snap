// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas LoginScreen debe mostrar component correctamente 1`] = `
<div
  className="allHeighWidth rs-flex-box-grid rs-flex-box-grid-top rs-flex-box-grid-center"
>
  <div
    className="h-100 rs-flex-box-grid-item rs-flex-box-grid-item-0 rs-col rs-col-lg-14 rs-col-md-12 rs-hidden-sm rs-hidden-xs"
    role="gridcell"
  >
    <div
      className="h-100 custom-slider rs-carousel rs-carousel-placement-bottom rs-carousel-shape-dot"
    >
      <div
        className="rs-carousel-content"
      >
        <div
          className="rs-carousel-slider"
          onTransitionEnd={[Function]}
          style={
            Object {
              "transform": "translate3d(-0%, 0 ,0)",
              "width": "400%",
            }
          }
        >
          <div
            aria-hidden={false}
            className="rs-carousel-slider-item carousel1"
            key="slider-item0/.0"
            style={
              Object {
                "width": "25%",
              }
            }
          >
            <div
              className="v-a-middle allHeightWidth"
              justify="center"
            >
              <h4>
                 Heroes App 
              </h4>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="rs-carousel-slider-item carousel5"
            key="slider-item1/.1"
            style={
              Object {
                "width": "25%",
              }
            }
          >
            <div
              className="v-a-middle allHeightWidth"
              justify="center"
            >
              <h4>
                 Los héroes que todo el mundo conoce al alcance de un click 
              </h4>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="rs-carousel-slider-item carousel6"
            key="slider-item2/.2"
            style={
              Object {
                "width": "25%",
              }
            }
          >
            <div
              className="v-a-middle allHeightWidth"
              justify="center"
            >
              <h4>
                 Lo mejor de Marvel y DC 
              </h4>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="rs-carousel-slider-item carousel2"
            key="slider-item3/.3"
            style={
              Object {
                "width": "25%",
              }
            }
          >
            <div
              className="v-a-middle allHeightWidth"
              justify="center"
            >
              <h4>
                 Busca y encuentra las características de tu héroe favorito 
              </h4>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rs-carousel-toolbar"
      >
        <ul>
          <li
            className="rs-carousel-label-wrapper"
            key="label0"
          >
            <input
              checked={true}
              id="indicator__u0kgnlcjnl_0"
              name="indicator__u0kgnlcjnl_0"
              onChange={[Function]}
              type="radio"
              value={0}
            />
            <label
              className="rs-carousel-label"
              htmlFor="indicator__u0kgnlcjnl_0"
            />
          </li>
          <li
            className="rs-carousel-label-wrapper"
            key="label1"
          >
            <input
              checked={false}
              id="indicator__u0kgnlcjnl_1"
              name="indicator__u0kgnlcjnl_1"
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <label
              className="rs-carousel-label"
              htmlFor="indicator__u0kgnlcjnl_1"
            />
          </li>
          <li
            className="rs-carousel-label-wrapper"
            key="label2"
          >
            <input
              checked={false}
              id="indicator__u0kgnlcjnl_2"
              name="indicator__u0kgnlcjnl_2"
              onChange={[Function]}
              type="radio"
              value={2}
            />
            <label
              className="rs-carousel-label"
              htmlFor="indicator__u0kgnlcjnl_2"
            />
          </li>
          <li
            className="rs-carousel-label-wrapper"
            key="label3"
          >
            <input
              checked={false}
              id="indicator__u0kgnlcjnl_3"
              name="indicator__u0kgnlcjnl_3"
              onChange={[Function]}
              type="radio"
              value={3}
            />
            <label
              className="rs-carousel-label"
              htmlFor="indicator__u0kgnlcjnl_3"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="rs-flex-box-grid-item rs-flex-box-grid-item-0 rs-col rs-col-lg-10 rs-col-md-12 rs-col-xs-22"
    role="gridcell"
  >
    <div
      className="container mt-5"
    >
      <div
        className="rs-flex-box-grid rs-flex-box-grid-top rs-flex-box-grid-center"
      >
        <div
          className="rs-flex-box-grid-item rs-flex-box-grid-item-0 rs-col rs-col-xs-24"
          role="gridcell"
        >
          <img
            alt="logo"
            className="imgStyle"
            src="logo.png"
          />
        </div>
        <div
          className="mt-3 rs-flex-box-grid-item rs-flex-box-grid-item-0 rs-col rs-col-xs-24"
          role="gridcell"
        >
          <h4
            className="textCenter"
            id="id_bienvenida"
          >
            Bienvenido a Heroes App
          </h4>
        </div>
        <div
          className="mt-5 rs-flex-box-grid-item rs-flex-box-grid-item-0 rs-col rs-col-md-12 rs-col-sm-18 rs-col-xs-20"
          role="gridcell"
        >
          <form
            className="rs-form rs-form-vertical rs-form-fluid"
            onSubmit={[Function]}
          >
            <div
              className="rs-form-group"
              role="group"
            >
              <label
                className="rs-form-control-label"
                id={null}
              >
                Email
              </label>
              <div
                className="rs-form-control rs-form-control-wrapper"
              >
                <input
                  aria-describedby={null}
                  aria-labelledby={null}
                  className="rs-input"
                  id="id_email_input"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  type="text"
                  value="juan1@gmail.com"
                />
              </div>
              <span
                className="rs-form-help-text"
                id={null}
              />
            </div>
            <div
              className="rs-form-group"
              role="group"
            >
              <label
                className="rs-form-control-label"
                id={null}
              >
                Password
              </label>
              <div
                className="rs-form-control rs-form-control-wrapper"
              >
                <input
                  aria-describedby={null}
                  aria-labelledby={null}
                  autoComplete="off"
                  className="rs-input"
                  id="id_password_input"
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  type="password"
                  value="xdlol1234"
                />
              </div>
              <span
                className="rs-form-help-text"
                id={null}
              >
                <span
                  className="forgotDiv"
                >
                  <button
                    className="rs-btn rs-btn-link rs-btn-xs"
                    type="button"
                  >
                    ¿Olvidaste tu contraseña?
                  </button>
                </span>
              </span>
            </div>
            <div
              className="mt-5 rs-btn-toolbar"
              role="toolbar"
            >
              <button
                className="centerButton rs-btn rs-btn-primary rs-btn-block"
                type="submit"
              >
                Submit
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                      "marginLeft": "10px",
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m21.426 11.095-17-8A.999.999 0 0 0 3.03 4.242L4.969 12 3.03 19.758a.998.998 0 0 0 1.396 1.147l17-8a1 1 0 0 0 0-1.81zM5.481 18.197l.839-3.357L12 12 6.32 9.16l-.839-3.357L18.651 12l-13.17 6.197z"
                    key="0"
                  />
                </svg>
                <span
                  className="rs-ripple-pond"
                >
                  <span
                    className="rs-ripple"
                  />
                </span>
              </button>
            </div>
          </form>
        </div>
        <div
          className="mt-4 rs-flex-box-grid-item rs-flex-box-grid-item-0 rs-col rs-col-md-13 rs-col-sm-18 rs-col-xs-20"
          role="gridcell"
        >
          <div
            aria-orientation="horizontal"
            className="rs-divider rs-divider-horizontal rs-divider-with-text"
            role="separator"
          >
            <span
              className="rs-divider-inner-text"
            >
               ó 
            </span>
          </div>
        </div>
        <div
          className="mt-4 googleDiv rs-flex-box-grid-item rs-flex-box-grid-item-0 rs-col rs-col-xs-24"
          role="gridcell"
        >
          <button
            className="rs-btn rs-btn-link"
            type="button"
          >
            <svg
              enableBackground="new 0 0 48 48"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              version="1.1"
              viewBox="0 0 48 48"
              width="1em"
              x="0px"
              xmlns="http://www.w3.org/2000/svg"
              y="0px"
            >
              <path
                d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
                fill="#FFC107"
                key="0"
              />
              <path
                d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
                fill="#FF3D00"
                key="1"
              />
              <path
                d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
                fill="#4CAF50"
                key="2"
              />
              <path
                d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
                fill="#1976D2"
                key="3"
              />
            </svg>
            Sign with Google
          </button>
        </div>
        <div
          className="mt-4 googleDiv rs-flex-box-grid-item rs-flex-box-grid-item-0 rs-col rs-col-xs-24"
          role="gridcell"
        >
          <span
            className="newAccount"
          >
            Nuevo en Heroes App?
            <button
              className="rs-btn rs-btn-link rs-btn-sm"
              type="button"
            >
              Crea una cuenta
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
